source/grid/grid_tools.cc

```diff
@@ -4209,34 +4246,75 @@ namespace GridTools
     std::pair<typename Triangulation<dim, spacedim>::active_cell_iterator,
               Point<dim>>
       my_pair;
+
+    bool         not_found      = true;
+    unsigned int points_checked = 0;
     if (cell_hint.state() == IteratorState::valid)
-      my_pair =
-        GridTools::find_active_cell_around_point(cache, points[0], cell_hint);
-    else
-      my_pair = GridTools::find_active_cell_around_point(cache, points[0]);
+      try
+        {
+          my_pair   = GridTools::find_active_cell_around_point(cache,
+                                                             points[0],
+                                                             cell_hint);
+          not_found = false;
+          ++points_checked;
+        }
+      catch (const GridTools::ExcPointNotFound<dim> &)
+        {
+          points_outside.emplace_back(0);
+          ++points_checked;
```

Comment:
The code would probably be easier to read if you moved the `++points_checked` line outside the `try` and `catch` blocks to after the whole construct. That's because this line will always be executed, either in the `try` or the `catch` section.